pkg/agent/stats/collector.go

```diff
@@ -250,12 +237,19 @@ func (m *Collector) report(curStatsCollection *statsCollection) error {
 		anpStats = mergeReportStats(multicastANPStatsMap, anpStats)
 	}
 
-	if len(npStats) == 0 && len(acnpStats) == 0 && len(anpStats) == 0 && !multicastGroupsUpdated {
-		klog.V(4).Info("No stats to report, skip reporting")
+	if len(npStats) == 0 && len(acnpStats) == 0 && len(anpStats) == 0 && isIdenticalMulticastGroupMap(m.lastStatsCollection.multicastGroups, curStatsCollection.multicastGroups) {
```

Comment:
With your current implementation, length of anpStats/acnpStats would never be 0 as long as multicastANPStatsMap/multicastACNPStatsMap have values, even if they are not changed in this report cycle?